src/core/prometheus.cc

```diff
@@ -896,16 +896,16 @@ class metrics_handler : public httpd::handler_base  {
         sstring metric_family_name = req->get_query_param("__name__");
         bool prefix = trim_asterisk(metric_family_name);
         bool show_help = req->get_query_param("__help__") != "false";
-        bool disable_aggregation = req->get_query_param("__aggregate__") != "false";
+        bool enable_aggregation = req->get_query_param("__aggregate__") != "false";
```

Comment:
Please note that you could have fixed this bug by a one-line change - changing != to ==, and that's it. The renaming (and flipping) the variable wasn't necessary. But I agree it looks better.

Reply:
i knew. but i am not after a single-liner fix.